offer-manager-domain/src/main/groovy/life/qbic/business/products/Product.groovy

```diff
@@ -0,0 +1,124 @@
+package life.qbic.business.products
+
+import life.qbic.datamodel.dtos.business.ProductCategory
+import life.qbic.datamodel.dtos.business.ProductId
+import life.qbic.datamodel.dtos.business.services.ProductUnit
+
+import java.nio.charset.StandardCharsets
+import java.security.MessageDigest
+
+/**
+ * <h1>Represents the product business model</h1>
+ * <br>
+ * <p>This class should be used in the business context of product creation</p>
+ *
+ * @since 1.0.0
+ *
+*/
+class Product {
+    private ProductCategory category
+    private String name
+    private String description
+    private double unitPrice
+    private ProductUnit unit
+    private ProductId id
+
+    static class Builder{
+        ProductCategory category
+        String name
+        String description
+        double unitPrice
+        ProductUnit unit
+        ProductId id
+
+        Builder(ProductCategory category, String name, String description, double unitPrice, ProductUnit unit){
+            this.category = Objects.requireNonNull(category)
+            this.name = Objects.requireNonNull(name)
+            this.description = Objects.requireNonNull(description)
+            this.unitPrice = Objects.requireNonNull(unitPrice)
+            this.unit = Objects.requireNonNull(unit)
+        }
+
+        Builder id(ProductId id){
+            this.id = id
+            return this
+        }
+
+        Product build(){
+            return new Product(this)
+        }
+    }
+
+    Product(Builder builder){
+        this.category = builder.category
+        this.name = builder.name
+        this.description = builder.description
+        this.unitPrice = builder.unitPrice
+        this.unit = builder.unit
+    }
+
+    /**
+     * Calculates the checksum for the product
+     * @return
+     */
+    String checksum(){
```

Comment:
Uncomplete javadocs. Since this is part of the object's API, it will help to know on which properties the checksum is computed on, won't it?